packages/unenv-preset/src/runtime/node/tls.ts

```diff
@@ -40,6 +38,45 @@ const workerdTls = process.getBuiltinModule("node:tls");
 // Natively implemented in workerd
 export const { connect, TLSSocket } = workerdTls;
 
+// `connect` implementation only supports a default SecureContext for now.
+export const createSecureContext: typeof nodeTls.createSecureContext = (
+	options?: nodeTls.SecureContextOptions
+) => {
+	const keys: (keyof nodeTls.SecureContextOptions)[] = [
+		"ALPNCallback",
+		"allowPartialTrustChain",
+		"ca",
+		"cert",
+		"sigalgs",
+		"ciphers",
+		"clientCertEngine",
+		"crl",
+		"dhparam",
+		"ecdhCurve",
+		"honorCipherOrder",
+		"key",
+		"privateKeyEngine",
+		"privateKeyIdentifier",
+		"maxVersion",
+		"minVersion",
+		"passphrase",
+		"pfx",
+		"secureOptions",
+		"secureProtocol",
+		"ticketKeys",
+		"sessionTimeout",
+	];
+
+	for (const k of keys) {
+		if (options?.[k] !== undefined) {
```

Comment:
Can we also omit null values?

```suggestion
		if (options?.[k] != null) {
```